### ISSUE TITLE
Script is broken - need a new backend solution, preferably free

### ISSUE THREAD

**Issue by WhatIsThisImNotGoodWithComputers** [flaghunters/Extra-Flags-for-int-#19]
Second host appears to be a trap to lure people into buying premium
![2015-01-02--1420228397_541x318_scrot](https://cloud.githubusercontent.com/assets/2182943/5599158/7c18a3e6-92c2-11e4-9f65-3b957502903a.png)
CPU limit reached after only a days use.

Need an alternative backend, these free hosts probably won't cut it.

**Comment by loadletter** [comment]
There's heroku, I had a thing that generated at every view 200 captchas and a scraper that runs continuously and never had any issue, but that was using python.

Limits are  2TB/month IIRC, and for languages that keep a script running it shuts down if not used after around 2 hours (to restart it takes a couple of seconds), but that shouldn't be a problem with enough users.

The only problem are the databases that have some heavy limitations (10K row limit on postgresql), maybe it could be set to connect to another hosting's database or doing hacks using arrays.

Also it requires using git, instead of FTP like normal hosting

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
I actually signed up for Heroku yesterday. Their free tier was looking bretty good, but it was quite a hassle to put up some simple php files. Also tried pagodabox which is something similar.

Would you happen to know any regex? I want to go with google forms.
I've created a simple form which takes board, post_nr and region.
This is posted to a google spreadsheet. 

I can currently http post to google forms, I just need to figure out the regex for only allowing
two or three characters for a boardname (abcdefghijklmnopqrstuvwxyz as whitelisted characters) and only characters from this (ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz,-) with a max of about 100 or so.

And I need to figure out how to get a single row from the spreadsheet.

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
This will be used for retrieval https://chriszarate.github.io/sheetrock/

**Comment by loadletter** [comment]
This should work

Board
```
[a-z]{2,3}
```

Text
```
[\w,-]{1,100}
```

**Comment by loadletter** [comment]
Also for retriaval this seems more updated https://github.com/jsoma/tabletop

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
I've changed the second regex to
```
[A-Za-z,-]{1,100}
```
to prevent numbers in there.

Does tabletop support search? sheetrock supports sql like searching.
It should support the new google spreadsheets.
They also mention it might get sluggish when rows hit 5000.
So we'll have to wait and see if the performance is affected after that.
Might do some caching or something else, but they do mention the lag is google api sided so browsers shouldn't slow, just the flags should load a few seconds later.

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
Would you be willing to test?
I've made a spreadsheet under a new account and my main gmail shows up as anonymous to it when I view the form but my main does see the flag hunters user.

Form here:
https://docs.google.com/forms/d/1KefKuG2b-8SMk38HgePE1mG1OazJrViRXkddsuG5O14/viewform?fbzx=7584299402871094819

Results here:
https://docs.google.com/spreadsheets/d/1fFDqDz7C3k8K-Yw6u2DPBHU-h5jUR4jp436ITc5Z8wU/edit#gid=1740830420

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
HTTP post url:
https://docs.google.com/forms/d/1KefKuG2b-8SMk38HgePE1mG1OazJrViRXkddsuG5O14/formResponse (application/x-www-form-urlencoded)
Post data (example):
```
entry.2089233394=fds&entry.808470227=1&entry.2015905474=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-,
```
I see anonymous chipmunk has joined my session so the forms truly is anonymous.
The forms doesn't appear to leak either, new changes are made by the sheet thingy which shows up as anonymous (Form responses).

**Comment by loadletter** [comment]
Oh, right you're going to load all of that everytime so tabletop can't be used

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
So posting a flag will be easy, now we'll just have to figure out a library for retrieval and we'll be set for another while (hopefully).
I don't have much time so I'll start on it tomorrow.

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
I wonder actually how sheetrock works, if it loads the whole thing and then applies filtering or filtering happens at google's side.

**Comment by loadletter** [comment]
Google side

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
Than few second delay is no problem at all.

**Comment by WhatIsThisImNotGoodWithComputers** [comment]
I signed up with Digital Ocean, got $10 free credit to get us through two months, after that $100 should be added because I signed up with the Github Educational deal.
Someone in neder gave me this tip, the performance will probably be better than Google Docs.
DO gives 1 core and 512gb ram with 1TB of data and 20GB SSD which should be plenty for the script.
Also DO is a reliable company so that should be okay too. 99.99% uptime.
Also, we won't have to change much code.
